### ISSUE TITLE
Fails to load due to sudo giving up errors about time of the day

### ISSUE THREAD

**Issue by morgenst3rn** [ngosang/docker-amule#3]
Everything works until the main executable is called with sudo on last line of entrypoint script

sudo -H -u '#'"${AMULE_UID}" sh -c "amuled -c ${AMULE_HOME} -o"

that gives this error:

sudo: unable to get time of day: Operation not permitted
sudo: error initializing audit plugin sudoers_audit

**Comment by ngosang** [comment]
* Are you using the image from DockerHub or building the image yourself?
* What is your OS and processor architecture?

**Comment by morgenst3rn** [comment]
I'm using the image from DockerHub, i'm running on a Raspbian GNU/Linux 10 (buster), Raspberry Pi 4 si it's ARM

**Comment by ngosang** [comment]
I'm running the same Docker image in a Raspberry Pi 4 (with Raspbian 11). It must be some bad configuration.

Could you try to run the `docker run` command with few parameters and see if it starts?
 `docker run ngosang/amule`

https://github.com/ngosang/docker-amule#docker-cli

**Comment by ngosang** [comment]
Closing due to no activity.

**Comment by morgenst3rn** [comment]
Sorry for the long delay, I've been unable to work on this issue for some time.
I've figured out that I've got some serious problems with ownership and permission of NAS folders, I've screwed up with some NFS mounts around.
Now I've fixed everything I'm able to start amule running the correct users, but I'm having issues with amule connection.
From the logs, everything seems in order (there is a small error about wxWidgets, don't know if could be relevant) 

> Creating group amule with GID 1000 ...
> Creating user amule with UID 1000 ...
> /home/amule/.aMule/amule.conf file found. Using existing configuration.
> /home/amule/.aMule/remote.conf file found. Using existing configuration.
>  15580-03-18 06:39:12: Logging to stdout enabled
>  9455-08-13 00:10:56: Initialising aMuleD 2.3.3 compiled with wxBase(GTK2) v3.0.4 and Boost 1.77
>  8230-09-12 00:10:56: Checking if there is an instance already running...
>  8230-09-12 00:10:56: No other instances are running.
>  16941-03-25 06:39:12: Creditfile loaded, 0 clients are known
>  6461-05-14 00:10:56:  - Credits expired for 6 clients!
>  8775-02-08 00:10:56: *** TCP socket (ECServer) listening on 0.0.0.0:34712
>  9183-05-31 00:10:56: Created Server UDP-Socket at port 34665
>  8230-09-12 00:10:56: ListenSocket: Ok.
>  8775-02-08 00:10:56: Created Client UDP-Socket at port 34672
>  17757-11-04 06:39:12: Loading server.met file: /home/amule/.aMule/server.met
> !15580-03-18 06:39:12: 19 servers in server.met found
> !17757-11-04 06:39:12: No server list address entry in 'addresses.dat' found. Please paste a valid server list address into this file in order to auto-update your server list
>  8230-09-12 00:10:56: Loading temp files from /temp.
>  8775-02-08 00:10:56: Loading PartFile 1 of 12
>  8775-02-08 00:10:56: Loading PartFile 2 of 12
>  8775-02-08 00:10:56: Loading PartFile 3 of 12
>  8775-02-08 00:10:56: Loading PartFile 4 of 12
>  8775-02-08 00:10:56: Loading PartFile 5 of 12
>  8775-02-08 00:10:56: Loading PartFile 6 of 12
>  8775-02-08 00:10:56: Loading PartFile 7 of 12
>  8775-02-08 00:10:56: Loading PartFile 8 of 12
>  8775-02-08 00:10:56: Loading PartFile 9 of 12
>  8775-02-08 00:10:56: Loading PartFile 10 of 12
>  8775-02-08 00:10:56: Loading PartFile 11 of 12
>  8775-02-08 00:10:56: Loading PartFile 12 of 12
>  8230-09-12 00:10:56: All PartFiles Loaded.
>  17621-09-28 06:39:12: Found 12 part files
>  17621-09-28 06:39:12: Adding file /temp/013.part.met to shares
>  17621-09-28 06:39:12: Adding file /temp/014.part.met to shares
>  17621-09-28 06:39:12: Adding file /temp/015.part.met to shares
>  17621-09-28 06:39:12: Adding file /temp/016.part.met to shares
>  17621-09-28 06:39:12: Adding file /temp/017.part.met to shares
>  17621-09-28 06:39:12: Adding file /temp/018.part.met to shares
>  17621-09-28 06:39:12: Adding file /temp/019.part.met to shares
>  17621-09-28 06:39:12: Adding file /temp/020.part.met to shares
>  17621-09-28 06:39:12: Adding file /temp/021.part.met to shares
>  17621-09-28 06:39:12: Adding file /temp/022.part.met to shares
>  17621-09-28 06:39:12: Adding file /temp/023.part.met to shares
>  17621-09-28 06:39:12: Adding file /temp/024.part.met to shares
>  15988-07-07 06:39:12: No shareable files found in directory: /incoming
>  17621-09-28 06:39:12: Found 12 known shared files
>  17485-08-21 06:39:12: amuled: OnInit - starting timer
> 15:49:36: Debug: In file /home/buildozer/aports/community/wxgtk/src/wxWidgets-3.0.4/src/unix/threadpsx.cpp at line 507: 'pthread_cond_timedwait()' failed with error 0x00000016 (Invalid argument).
>  15580-03-18 06:39:12: Asio thread 1 started
>  15580-03-18 06:39:12: Asio thread 2 started
>  15580-03-18 06:39:12: Asio thread 3 started
>  15580-03-18 06:39:12: Asio thread 4 started
>  8230-09-12 00:10:56: Loading IP filters 'ipfilter.dat' and 'ipfilter_static.dat'.
>  8230-09-12 00:10:56: Loaded 0 IP-ranges from '/home/amule/.aMule/ipfilter.dat'.
>  8230-09-12 00:10:56: Loaded 0 IP-ranges from '/home/amule/.aMule/ipfilter_static.dat'.
>  18438-05-09 06:39:12: IP filter is ready
> !15988-07-07 06:39:12: Connecting
>  8230-09-12 00:10:56: Connecting to eMule Security (5.45.85.226 - 5.45.85.226:6584) using protocol obfuscation.
>  17621-09-28 06:39:12: Read 153 Kad contacts
> !8230-09-12 00:10:56: Kad started.
>  7550-03-10 00:10:56: Connected to eMule Security (5.45.85.226:6584)
>  15580-03-18 06:39:12: New external connection accepted
>  8639-01-02 00:10:56: Connecting client: amule-remote 0x0001
>  8230-09-12 00:10:56: Access granted.


![2022-02-25_123413](https://user-images.githubusercontent.com/42541/155709111-fec083f3-01d8-46e5-9569-5502ec1f6842.png)
![2022-02-25_123440](https://user-images.githubusercontent.com/42541/155709115-9e24624f-421f-4fd6-9083-423860a2a58d.png)
![2022-02-25_123449](https://user-images.githubusercontent.com/42541/155709116-be0f57e3-b047-41b2-a063-ff1d7127eda8.png)
![2022-02-25_123459](https://user-images.githubusercontent.com/42541/155709118-b3e86963-76cb-4ae2-add5-6f0c1ec40008.png)


But despite the log saying that it's connected, amule seems to be stuck with amber and yellow status, search is not working.
I've reviewed the ports and all is forwarded correctly, I even tried another docker with a less recent version of amule and given the same amule.conf it works, but since it's an old version it crashes too frequently, yours is way more stable.
Do you have any suggestions about where to look for this issue?

**Comment by ngosang** [comment]
* Try to remove `amule.conf` before starting the container. It will create a new file from scratch.
* Use the WebUI instead of the desktop client.
* Make sure the desktop client version matches aMule Docker version (2.3.3). I have tested 2.3.2 and it's working too.

**Comment by morgenst3rn** [comment]
Thanks for the suggestions. I've tried to, but it seems to no avail.
Had the conf recreated, didn't differ so much from mine, I've just customized port numbers and password hash for web and GUI clients. Connection with desktop client is still successful but with the same odd behavior as before.
With the WebUI, I can connect once but as soon as I enter the correct password I get a connection refused, log complain about amuleweb command exited with status code -6 


![image](https://user-images.githubusercontent.com/42541/160361715-d847f469-8ada-4812-a46c-53539e740f7a.png)
